app/controllers/user_mentee_application_cohorts_controller.rb

```diff
@@ -8,7 +8,24 @@ def index
 
   def show
     @cohort = UserMenteeApplicationCohort
-              .includes(:user_mentee_applications)
+              .includes(user_mentee_applications: :mentee_application_states)
               .find(params[:id])
+
+    @filtered_applications = filter_applications(@cohort.user_mentee_applications)
+  end
+
+  private
+
+  def filter_applications(applications)
+    if params[:filter]
+      case params[:filter]
+      when 'in_review'
+        applications.filter(&:in_review?)
+      else
+        applications.filter { |application| application.current_status == params[:filter] }
+      end
+    else
+      applications.filter(&:in_review?)
+    end
```

Comment:
I think the logic could be simpler here. Good to notice with conditionals when conditions or body expressions are being repeated, like with `applications.filter(&:in_review?)`

Could maybe have something like:
```rb
case params[:filter]
when 'accepted', 'rejected'
  applications.filter { |application| application.current_status == params[:filter] }
else
  applications.filter(&:in_review?)
end
```

Reply:
ah good call, that's definitely redundant logic! Thank you